File: septunneuf-serveur/septunneuf-monsters/src/main/java/fr/ebiron/septunneuf/monsters/handler/GlobalExceptionHandler.java
package fr.ebiron.septunneuf.monsters.handler;


import fr.ebiron.septunneuf.monsters.dto.HttpExceptionDto;
import fr.ebiron.septunneuf.monsters.exception.ConflictException;
import fr.ebiron.septunneuf.monsters.exception.NotFoundException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.ResponseStatus;

@ControllerAdvice
public class GlobalExceptionHandler {

    private final Logger log = LoggerFactory.getLogger(getClass());

    @ExceptionHandler(NotFoundException.class)
    @ResponseStatus(HttpStatus.NOT_FOUND)
    @ResponseBody
    public HttpExceptionDto handleNotFoundException(NotFoundException e) {
        log.error("NotFoundException: {}", e.getMessage());
        return new HttpExceptionDto(HttpStatus.NOT_FOUND.value(), HttpStatus.NOT_FOUND.getReasonPhrase(), e.getMessage());
    }

    @ExceptionHandler(ConflictException.class)
    @ResponseStatus(HttpStatus.CONFLICT)
    @ResponseBody
    public HttpExceptionDto handleConflictException(ConflictException e) {
        log.error("ConflictException: {}", e.getMessage());
        return new HttpExceptionDto(HttpStatus.CONFLICT.value(), HttpStatus.CONFLICT.getReasonPhrase(), e.getMessage());
    }
}

File: septunneuf-serveur/septunneuf-inventory/src/main/kotlin/fr/ebiron/septunneuf/inventory/handlers/GlobalExceptionHandler.kt
package fr.ebiron.septunneuf.inventory.handlers

import fr.ebiron.septunneuf.inventory.dto.HttpExceptionDto
import fr.ebiron.septunneuf.inventory.exceptions.MonsterInventoryNotFound
import org.slf4j.LoggerFactory
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.ResponseBody
import org.springframework.web.bind.annotation.ResponseStatus

@ControllerAdvice
class GlobalExceptionHandler {

    private val log = LoggerFactory.getLogger(javaClass)

    @ExceptionHandler(MonsterInventoryNotFound::class)
    @ResponseStatus(HttpStatus.NOT_FOUND)
    @ResponseBody
    fun handleNotFoundException(e: MonsterInventoryNotFound): HttpExceptionDto {
        log.error("NotFoundException: ${e.message}")
        return HttpExceptionDto(HttpStatus.NOT_FOUND.value(), HttpStatus.NOT_FOUND.reasonPhrase, e.message)
    }
}